packages/material-ui-styles/src/useThemeVariants/useThemeVariants.test.js

```diff
@@ -0,0 +1,97 @@
+import { expect } from 'chai';
+import React from 'react';
+import { createClientRender, screen } from 'test/utils';
+import { createMuiTheme } from '@material-ui/core/styles';
+import ThemeProvider from '../ThemeProvider';
+import useThemeVariants from './useThemeVariants';
+import withStyles from '../withStyles';
+
+describe('useThemeVariants', () => {
+  const render = createClientRender();
+
+  const ComponentInternal = (props) => {
+    const { className, ...other } = props;
+    const themeVariantsClasses = useThemeVariants(props, 'Test');
+    return <div className={`${themeVariantsClasses} ${className}`} {...other} />;
+  };
+
+  const Component = withStyles({}, { name: 'Test' })(ComponentInternal);
+
+  it('returns variants classes if props are match', () => {
+    const theme = createMuiTheme({
+      variants: {
+        Test: [
+          {
+            props: { variant: 'test' },
+            styles: { backgroundColor: 'rgb(255, 0, 0)' },
+          },
+        ],
+      },
+    });
+
+    render(
+      <ThemeProvider theme={theme}>
+        <Component data-testid="component" variant="test">
+          Test
+        </Component>
+      </ThemeProvider>,
+    );
+
+    const style = window.getComputedStyle(screen.getByTestId('component'));
+    expect(style.getPropertyValue('background-color')).to.equal('rgb(255, 0, 0)');
+  });
+
+  it('does not return variants classes if props are match', () => {
```

Comment:
```suggestion
  it('does not return variants classes if props do not match', () => {
```
? 

Otherwise this is the same condition as above but with a different result. It's unclear why the same inputs return different results from the description.